### ISSUE TITLE
[BUG] salt minion key in denied list after using salt-call command on the minion

### ISSUE THREAD

**Issue by wesley-c42** [saltstack/salt#59418]
**Description**
One of my salt minions has its key placed in the denied list as soon as I use a salt-call command (in my case `test.ping`). I can still run salt commands to the minion from the master.
When running a salt-call command I get the following message:
```
[CRITICAL] The Salt Master has rejected this minion's public key!
To repair this issue, delete the public key for this minion on the Salt Master and restart this minion.
Or restart the Salt Master in open mode to clean out the keys. The Salt Minion will now exit.
```

The minion's key is also still in the accepted keys list, I've checked both keys, they are identical.
When I remove the denied key (or both the accepted and denied key), restart the minion and accept its key again, all seems fine. As soon as I run a salt-call command on the minion again though, I get the error message as above and it's key also shows up in the denied list. I can at this point still run salt commands to the minion from the master.

**Setup**
Both master and minion running 3000.5 and ubuntu 18.04.

**Versions Report**
```
Salt Version:
           Salt: 3000.5

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: 2.0.3
      gitpython: 2.1.8
         Jinja2: 2.10
        libgit2: 0.26.0
       M2Crypto: Not Installed
           Mako: 1.0.7
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: 0.26.2
         Python: 2.7.17 (default, Sep 30 2020, 13:38:04)
   python-gnupg: 0.4.1
         PyYAML: 3.12
          PyZMQ: 16.0.2
          smmap: 2.0.3
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.2.5

System Versions:
           dist: Ubuntu 18.04 bionic
         locale: UTF-8
        machine: x86_64
        release: 4.15.0-134-generic
         system: Linux
        version: Ubuntu 18.04 bionic
```

**Comment by wesley-c42** [comment]
So I just saw this happening on another minion. I've tried removing keys on both master and minion side and re-authenticating, but this did not work. Purging and re-installing did work.
About possible cause; I'm not 100% sure anymore about the first minion, but on this last one definitely, I had changed the minion ID after authenticating. It is the only issue related thing I can think of that is different from all my other minions where I've not seen this issue.

**Comment by sagetherage** [comment]
@wesley-c42
The Salt version is vulnerable. We recommend upgrading as soon as possible to at least latest point release or to latest. You can find those here https://github.com/saltstack/salt/releases or the packages repo here: https://repo.saltproject.io/

I also believe you will not see this issue if you upgrade to the latest point version v3000.9.

**Comment by sagetherage** [comment]
@wesley-c42 
Have you been able to upgrade or are you still encountering this issue?

**Comment by garethgreenaway** [comment]
@wesley-c42 Following up on this one.  Were you able to upgrade to the latest Salt version which is now 3003.1?

**Comment by wesley-c42** [comment]
@sagetherage @garethgreenaway Not yet, I will fast track this though as I am encountering this issue the other way around now; salt-call commands on minions are working correctly, but salt commands to minions from master are returning `[Not connected]`. I will update once I have.

What's also strange to me is that during all this there is no salt process running and the salt-minion service is in error (yet salt-call commands still work):
```
root@server1:~# ps -ef |grep salt
root     26566 18727  0 14:25 pts/1    00:00:00 grep --color=auto salt
root@server1:~# service salt-minion status
● salt-minion.service - The Salt Minion
   Loaded: loaded (/lib/systemd/system/salt-minion.service; enabled; vendor preset: enabled)
   Active: inactive (dead) since Wed 2021-07-21 08:35:18 UTC; 1 day 5h ago
     Docs: man:salt-minion(1)
           file:///usr/share/doc/salt/html/contents.html
           https://docs.saltstack.com/en/latest/contents.html
  Process: 1393 ExecStart=/usr/bin/salt-minion (code=exited, status=0/SUCCESS)
 Main PID: 1393 (code=exited, status=0/SUCCESS)

Jul 21 08:35:15 server1 systemd[1]: Starting The Salt Minion...
Jul 21 08:35:16 server1 systemd[1]: Started The Salt Minion.
Jul 21 08:35:18 server1 salt-minion[1393]: [CRITICAL] The Salt Master has rejected this minion's public key!
Jul 21 08:35:18 server1 salt-minion[1393]: To repair this issue, delete the public key for this minion on the Salt Master and restart this minion.
Jul 21 08:35:18 server1 salt-minion[1393]: Or restart the Salt Master in open mode to clean out the keys. The Salt Minion will now exit.
root@server1:~# salt-call test.ping
local:
    True
```

**Comment by sagetherage** [comment]
@wesley-c42 verify that the minion is not running then try:
clearing out the pki directory on the minion and on the master deleting the key using salt-key and see if that solves the issue.

**Comment by garethgreenaway** [comment]
@wesley-c42 We noticed that one of your previous comments indicated that you had changed the ID for the minion after the authentication process, this would definitely cause the errors that you're seeing as it would cause the Salt master to see a connection from a different minion and reject the connection because the existing keys do not match the new minion ID.   Anytime you rename the ID of the minion it's best practice to clear out the `pki` directory on the Salt minion and delete any existing keys for that minion on the Salt master using the `salt-key` tool.

**Comment by garethgreenaway** [comment]
Closing this one out due to the lack of activity.  If the issue persists please feel free to open a new issue.

**Comment by wesley-c42** [comment]
Sorry for the late response. In the meantime I have upgraded to version 3003.2 and have not seen the issue occur. Before that I also tried the intermediate update to 3000.9, but still saw the issue occur.

When running on the older version is did clear out the pki directories multiple times, but the issue kept occurring.

Anyway, upgrading to the latest version seems to have fixed it. Before it was easily reproducible at least, but now it is not.
Thanks for the help!
